misc/scripts/library-coverage/comment-pr.py

```diff
@@ -95,14 +110,37 @@ def comment_pr(repo, run_id):
 
         comment = comment_first_line + \
             "A recent commit removed the previously reported differences."
-    post_comment(comment, repo, pr_number)
+
+    if raw_comment_id is None:
+        post_initial_comment(comment, repo, pr_number)
+    else:
+        update_existing_comment(comment, repo, pr_number, raw_comment_id)
 
 
-def post_comment(comment, repo, pr_number):
+def post_initial_comment(comment, repo, pr_number):
     print(f"Posting comment to PR #{pr_number}")
     utils.subprocess_run(["gh", "pr", "comment", str(pr_number),
                          "--repo", repo, "--body", comment])
 
+def update_existing_comment(comment, repo, pr_number, raw_comment_id):
+    # Fetch existing comment, and validate:
+    # - comment belongs to the PR with number `pr_number`
+    # - comment starts with the expected prefix `comment_first_line`
+    # - comment author is github-actions[bot]
+    comment_author = "github-actions[bot]"
+    filter = f"select(.issue_url | endswith(\"{repo}/issues/{pr_number}\")) " + \
```

Comment:
Does the comments API not return an issue/PR number in a separate field? Parsing the issue URL to get the number is...correct but unfortunate if we have to do it :)

Reply:
I'm not seeing an issue/PR number in a separate field when I review the [API docs](https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28) about issue comments. It seems that the issue/PR number needs to be parsed from either the `issue_url` or the `html_url`. 
(I'm fairly new to using the API though, so let me know if you think I missed something.)

Reply:
That is an annoying limitation of the API 😆 . What you have looks best then. Another idea would be to store the PR number in an artifact too, but that feels like too much.